components/camel-amqp/src/main/java/org/apache/camel/component/amqp/AMQPConfiguration.java

```diff
@@ -51,4 +64,28 @@ public void setIncludeAmqpAnnotations(boolean includeAmqpAnnotations) {
         this.includeAmqpAnnotations = includeAmqpAnnotations;
     }
 
+    public static class CamelAmqpTemplate extends CamelJmsTemplate {
+
+        public CamelAmqpTemplate(AMQPConfiguration config, ConnectionFactory connectionFactory) {
+            super(config, connectionFactory);
+        }
+
+        @Override
+        protected void commitIfNecessary(Session session) throws JMSException {
+            //# CAMEL-20521 - we can't use JmsUtils.commitIfNecessary because it catches
+            // jakarta.jms.IllegalStateException that can be thrown if the session is closed
```

Comment:
I'm not sure why this can fix CAMEL-20521 when session is closed during the commit? It will lead to the re-connecting?

Reply:
It leads to a redelivery if configured, this fix allows the exception to bubble up and can be handled by the error handler.
At the moment the exception is stopped at   `org.springframework.jms.support.JmsUtils.commitIfNecessary`, not even logged at trace level, so the camel route continue on the normal flow, like the commit has ended successfully.

Reply:
OK, thanks for the explaination!